src/middlewared/middlewared/plugins/netdata.py

```diff
@@ -0,0 +1,241 @@
+
+            for bind_ip in bind_to_ips:
+                if bind_ip not in valid_ips:
+                    verrors.add(
+                        'netdata_update.bind',
+                        f'Invalid {bind_ip} bind IP'
+                    )
+        else:
+            verrors.add(
+                'netdata_update.bind',
+                'This field is required'
+            )
+
+        update_alarms = data.pop('update_alarms', {})
+        valid_alarms = await self.middleware.run_in_thread(self.list_alarms)
+        if update_alarms:
+            for alarm in update_alarms:
+                if alarm not in valid_alarms:
+                    verrors.add(
+                        'netdata_update.alarms',
+                        f'{alarm} not a valid alarm'
+                    )
+
+            verrors.extend(
+                validate_attributes([Bool(key) for key in update_alarms], {'attributes': update_alarms})
+            )
+
+        # Validating streaming metrics now
+        stream_mode = data.get('stream_mode')
+        if stream_mode == 'SLAVE':
+            for key in ('api_key', 'destination'):
+                if not data.get(key):
+                    verrors.add(
+                        f'netdata_update.{key}',
+                        f'{key} is required with stream mode as SLAVE'
+                    )
+
+            destinations = data.get('destination')
+            if destinations:
+                ip_addr = IpAddress()
+                port = Port()
+                for dest in destinations:
+                    ip = dest.split(':')[0]
+                    try:
+                        ip_addr(ip)
+                    except ValueError as e:
+                        verrors.add(
+                            'netdata_update.destination',
+                            str(e)
+                        )
+                    else:
+                        if ':' in dest:
+                            try:
+                                port(dest.split(':')[1])
+                            except ValueError as e:
+                                verrors.add(
+                                    'netdata_update.destination',
+                                    f'Not a valid port: {e}'
+                                )
+        elif stream_mode == 'MASTER':
+            for key in ('allow_from', 'api_key'):
+                if not data.get(key):
+                    verrors.add(
+                        f'netdata_update.{key}',
+                        f'{key} is required with stream mode as MASTER'
+                    )
+
+        verrors.check()
+
+        for alarm in valid_alarms:
+            # Let's add alarms to our db if they aren't already there
+            if alarm not in data['alarms']:
+                data['alarms'][alarm] = True
+
+        for alarm in update_alarms:
+            # These are valid alarms
+            data['alarms'][alarm] = update_alarms[alarm]
```

Comment:
Wouldnt data['alarms'].update(update_alarms) be simpler/faster? Or did I miss something.

Reply:
You're right, it will be simpler/faster. Will have it updated asap : )